libclamav/ole2_extract.c

```diff
@@ -2635,13 +2947,26 @@ cl_error_t cli_ole2_extract(const char *dirname, cli_ctx *ctx, struct uniq **fil
         /* PASS 2/B : OTF scan */
         file_count = 0;
         if (bEncrypted) {
-            ret = ole2_walk_property_tree(&hdr, NULL, 0, handler_otf_encrypted, 0, &file_count, ctx, &scansize2, &key);
+            ret = ole2_walk_property_tree(&hdr, NULL, 0, handler_otf_encrypted, 0, &file_count, ctx, &scansize2, &key, &encryption_status);
         } else {
-            ret = ole2_walk_property_tree(&hdr, NULL, 0, handler_otf, 0, &file_count, ctx, &scansize2, NULL);
+            ret = ole2_walk_property_tree(&hdr, NULL, 0, handler_otf, 0, &file_count, ctx, &scansize2, NULL, &encryption_status);
         }
     }
 
 done:
+
+    if (encryption_status.encryption_type) {
+        insert_metadata(ctx, encryption_status.encryption_type, true);
+    }
+    insert_metadata(ctx, "Encrypted", encryption_status.encrypted);
+    insert_metadata(ctx, "EncryptedWithVelvetSweatshop", encryption_status.velvet_sweatshop);
+
+    if (encryption_status.velvet_sweatshop) {
+        print_heuristic(ctx, "Heuristics.Encrypted.OLE2.VelvetSweatshop");
+    } else if (encryption_status.encrypted) {
+        print_heuristic(ctx, "Heuristics.Encrypted.OLE2");
+    }
+
```

Comment:
I had thought about that, but it would change the VelvetSweatshop behavior from what we had in previous releases.  I am ok with that, but do you think anyone will complain?

Reply:
Per convo with Josh from THR, sounds like we could get away with an even bigger change and replace these three fields with one, along the lines of:
```json
"Encrypted": "VelvetSweatshop"
```
or
```json
"Encrypted": "RC4"
```
I realize this is a little silly since VelvetSweatshop is the password and not the encryption algorithm.  But it also indicates that it can be decrypted automatically whereas for RC4 we wouldn't know the password so it is "proper encrypted" and cannot be scanned. 
What do you think?  I'm open to more brainstorming if there's an even more elegant solution.

Reply:
The issue with moving all of that logic to before the 'done' label, is that it will not catch the cases where it's encrypted with AES128, but 'VelvetSweatshop' is not the password (fails the validation).  

I do agree with the behavior you are suggesting.

Reply:
I personally like 
"Encrypted":"Algorithm"
and if it's not VelvetSweatshop, adding the heuristic.  I do think that if there is a parse error or something other than CL_VIRUS, it is relevant to add the encryption type to the json metadata, but not the heuristic alert.  If you are ok with that, I'll add it.